pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -479,6 +486,59 @@ func (p *ConfigurationEditFlags) Apply(
 		service.Spec.Template.Spec.TimeoutSeconds = &p.TimeoutSeconds
 	}
 
+	if cmd.Flags().Changed("profile") {
+		profileName := ""
+		deleteProfile := false
+		if strings.HasSuffix(p.Profile, "-") {
+			profileName = p.Profile[:len(p.Profile)-1]
+			deleteProfile = true
+		} else {
+			profileName = p.Profile
+			deleteProfile = false
+		}
+
+		if len(knconfig.GlobalConfig.Profile(profileName).Annotations) > 0 || len(knconfig.GlobalConfig.Profile(profileName).Labels) > 0 {
+			annotations := knconfig.GlobalConfig.Profile(profileName).Annotations
+			labels := knconfig.GlobalConfig.Profile(profileName).Labels
+
+			profileAnnotations := make(util.StringMap)
+			for _, value := range annotations {
+				profileAnnotations[value.Name] = value.Value
+			}
+
+			profileLabels := make(util.StringMap)
+			for _, value := range labels {
+				profileLabels[value.Name] = value.Value
+			}
+
+			if deleteProfile {
+				var annotationsToRemove []string
+				for _, value := range annotations {
+					annotationsToRemove = append(annotationsToRemove, value.Name)
+				}
+
+				var labelsToRemove []string
+				for _, value := range labels {
+					labelsToRemove = append(labelsToRemove, value.Name)
+				}
+
+				err = servinglib.UpdateRevisionTemplateAnnotations(template, map[string]string{}, annotationsToRemove)
```

Comment:
I'd throw any error immediately, not proceeding with updates anymore. Depending how likely we can hit corner case here. Applicable to `else` branch as well.

```suggestion
				if err := servinglib.UpdateRevisionTemplateAnnotations(template, map[string]string{}, annotationsToRemove); err != nil {
			        return err
				} 
```